staging/prometheus-operator/dependent-charts/kube-state-metrics/templates/deployment.yaml

```diff
@@ -3,20 +3,26 @@ kind: Deployment
 metadata:
   name: {{ template "kube-state-metrics.fullname" . }}
   labels:
-    app: {{ template "kube-state-metrics.name" . }}
-    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
-    release: "{{ .Release.Name }}"
-    heritage: "{{ .Release.Service }}"
+    app.kubernetes.io/name: {{ template "kube-state-metrics.name" . }}
+    helm.sh/chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+    app.kubernetes.io/instance: "{{ .Release.Name }}"
+    app.kubernetes.io/managed-by: "{{ .Release.Service }}"
+{{- if .Values.customLabels }}
+{{ toYaml .Values.customLabels | indent 4 }}
+{{- end }}
 spec:
   selector:
     matchLabels:
-      app: {{ template "kube-state-metrics.name" . }}
+      app.kubernetes.io/name: {{ template "kube-state-metrics.name" . }}
```

Comment:
Im pretty sure this will cause issues when upgrading. I seem to remember that helm couldn't upgrade because you are not allowed to change the `selector` field.

Reply:
That is not right. The breaking changes applied not the the label selectors but to all the label changes. I thought you asked me to upgrade the kube-state-metrics ;)

Reply:
Therefore, I propose you two options:
1) keep the old labels and remove the new ones
2) keep both labels the old and new ones

Reply:
I'm actually not sure if this will prevent an upgrade, was more of a question. If there is an issue we would have to go with option 1.

Reply:
Just reread the options.
Option 2 makes sense to me.

Reply:
👍